JustSaying/AwsTools/MessageHandling/SqsNotificationListener.cs

```diff
@@ -237,7 +237,7 @@ private Task HandleMessage(Amazon.SQS.Model.Message message, CancellationToken c
             return _messageProcessingStrategy.StartWorker(action, ct);
         }
 
-        public ICollection<ISubscriber> Subscribers { get; set; }
+        public ICollection<ISubscriber> Subscribers { get; }
```

Comment:
Do we _actually_ want to do this to all these types?

Reply:
For `PublishMetadata` yes, I made that class recently and we can get by without a settor on the collection. 
For the other two interfaces 🤷‍♂️ if you wanted to keep it, shoulda put a test on it.

Reply:
Counter example: all "options" classes that use collections would violate this rule and then config binding would silently fail because the setter was no longe public.

Adding tests to test that an API surface for a simple POCO with a get and a set lets you get and set..._really_?

Reply:
This is the interface `INotificationSubscriberInterrogation` - what does that really do?

Reply:
I believe it's something to allow introspection on "stuff" after the listeners are all set up.

Reply:
The interface doesn't appear to be used anywhere, in internal code or on the public github https://github.com/search?q=INotificationSubscriberInterrogation&type=Code
It seems to only be used inside JustSaying

Reply:
Maybe instead we should just remove it instead?

Reply:
I mean, the `get` is used via `JustSayingBus.AddNotificationSubscriber`. The set appears to be unused.

Reply:
Well as long as it's actually being initialized and populated with stuff then.